mail_digest/models/res_partner.py

```diff
@@ -0,0 +1,209 @@
+    )
+
+    @api.multi
+    def _compute_notify_subtypes(self, enabled):
+        self.ensure_one()
+        query = (
+            'SELECT subtype_id FROM partner_notification_conf '
+            'WHERE partner_id=%s AND enabled = %s'
+        )
+        self.env.cr.execute(
+            query, (self.id, enabled))
+        return [x[0] for x in self.env.cr.fetchall()]
+
+    @api.multi
+    @api.depends('notify_conf_ids.subtype_id')
+    def _compute_enabled_notify_subtype_ids(self):
+        for partner in self:
+            partner.enabled_notify_subtype_ids = \
+                partner._compute_notify_subtypes(True)
+
+    @api.multi
+    @api.depends('notify_conf_ids.subtype_id')
+    def _compute_disabled_notify_subtype_ids(self):
+        for partner in self:
+            partner.disabled_notify_subtype_ids = \
+                partner._compute_notify_subtypes(False)
+
+    def _search_notify_subtype_ids_domain(self, operator, value, enabled):
+        if operator in ('in', 'not in') and \
+                not isinstance(value, (tuple, list)):
+            value = [value, ]
+        conf_value = value
+        if isinstance(conf_value, int):
+            # we search conf records always w/ 'in'
+            conf_value = [conf_value]
+        _value = self.env['partner.notification.conf'].search([
+            ('subtype_id', 'in', conf_value),
+            ('enabled', '=', enabled),
+        ]).mapped('partner_id').ids
+        return [('id', operator, _value)]
+
+    def _search_enabled_notify_subtype_ids(self, operator, value):
+        return self._search_notify_subtype_ids_domain(
+            operator, value, True)
+
+    def _search_disabled_notify_subtype_ids(self, operator, value):
+        return self._search_notify_subtype_ids_domain(
+            operator, value, False)
+
+    @api.multi
+    def _notify(self, message, force_send=False, user_signature=True):
+        """Override to delegate domain generation."""
+        # notify_by_email
+        email_domain = self._get_notify_by_email_domain(message)
+        # `sudo` from original odoo method
+        # the reason should be that anybody can write messages to a partner
+        # and you really want to find all ppl to be notified
+        partners = self.sudo().search(email_domain)
+        partners._notify_by_email(
+            message, force_send=force_send, user_signature=user_signature)
+        # notify_by_digest
+        digest_domain = self._get_notify_by_email_domain(
+            message, digest=True)
+        partners = self.sudo().search(digest_domain)
+        partners._notify_by_digest(message)
+
+        # notify_by_chat
+        self._notify_by_chat(message)
+        return True
+
+    @api.multi
+    def _notify_by_digest(self, message):
+        message_sudo = message.sudo()
+        if not message_sudo.message_type == 'email':
+            return
+        # TODO: what to do when a partner updates frequency?
```

Comment:
yep, you are right here. In your opinion, is there any other place where this should be taken into account?